logstash-plugins/logstash-input-tcp
# Standardize and add SSL settings

### What this PR does?
   
Reviewed and deprecated the following SSL settings to comply with Logstash's naming convention:
`ssl_enable` in favor of `ssl_enabled`
`ssl_cert` in favor of `ssl_certificate`
`ssl_verify` in favor of `ssl_client_authentication` when mode is `server`
`ssl_verify` in favor of `ssl_verification_mode` when mode is `client`

Added SSL configuration validations
 - Added a warning log when `ssl_*` configs are set but `ssl_enabled => false`
 - `ssl_ceritifcate` & `ssl_key` config validation, including checking if both files are readable.
 - Ensure no `ssl_client_authentication` when `mode => client`
 - Ensure no `ssl_verification_mode` when `mode => server`
 - Ensure an `ssl_certificate` is provided when `mode => server`
 - Ensure  `ssl_certificate_authorities` is provided when `mode => server` and it's using the new configuration (`ssl_client_authentication`)

The behavior standardization across plugins, such as the accepted certificate formats, default values, etc will be tackled in future PRs.

---
Closes: https://github.com/elastic/logstash/issues/14927
Closes: https://github.com/logstash-plugins/logstash-input-tcp/issues/42
